src/Views/AppListUpdateView.vala

```diff
@@ -67,6 +69,40 @@ namespace AppCenter.Views {
 
             add (infobar);
             add (scrolled);
+
+            refresh_cancellable = new Cancellable ();
+
+            get_apps.begin ();
+
+            unowned var client = AppCenterCore.Client.get_default ();
+            client.installed_apps_changed.connect (() => {
+                Idle.add (() => {
+                    get_apps.begin ();
+                    return GLib.Source.REMOVE;
+                });
+            });
+        }
+
+        private async void get_apps () {
+            refresh_cancellable.cancel ();
+
+            yield refresh_mutex.lock ();
+
+            refresh_cancellable.reset ();
```

Comment:
According to Valadoc.org:
> Note that it is generally not a good idea to reuse an existing cancellable for more operations after it has been cancelled once, as this function might tempt you to do. The recommended practice is to drop the reference to a cancellable after cancelling it, and let it die with the outstanding async operations. You should create a fresh cancellable for further async operations.

However, this is existing code so not necessarily blocking for this PR.